src/librustc/ty/context.rs

```diff
@@ -2568,7 +2573,7 @@ impl<'tcx> TyCtxt<'tcx> {
 
     #[inline]
     pub fn mk_self_type(self) -> Ty<'tcx> {
-        self.mk_ty_param(0, kw::SelfUpper.as_interned_str())
+        self.types.self_param
```

Comment:
Looks like `mk_self_type` isn't commonly used, perhaps it can be removed in favor of using `self.types.self_param` directly?